pandas-stubs/core/frame.pyi

```diff
@@ -138,17 +138,17 @@ class _LocIndexerFrame(_LocIndexer):
     @overload
     def __getitem__(
         self,
-        idx: Union[int, StrLike],
+        idx: Union[ScalarT, None],
     ) -> Series: ...
     @overload
     def __getitem__(
         self,
-        idx: Tuple[Union[IndexType, MaskType], StrLike],
+        idx: Tuple[Union[IndexType, MaskType], Union[ScalarT, None]],
     ) -> Series: ...
     @overload
     def __getitem__(
         self,
-        idx: Tuple[Tuple[slice, ...], StrLike],
+        idx: Tuple[Tuple[slice, ...], Union[ScalarT, None]],
```

Comment:
Now that I look at this, I think we could combine the 3 overloads here into one with the argument `Union[ScalarT, None, Tuple[Union[IndexType, MaskType], Union[ScalarT, None]], Tuple[Tuple[slice, ...], Union[ScalarT, None]]]`

I think this combination was just missed in the past.

Other issue in using `ScalarT` is whether it might be too wide.  Are we sure that all the types that are in `Scalar` can be used as a column name in a `DataFrame` ? (e.g., `Timestamp`, `Timedelta`, `datetime.date`, etc.)

Reply:
I managed to simplify it even more since the two `Tuple` types could be further compressed into a single `Tuple`